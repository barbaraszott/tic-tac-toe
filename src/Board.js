import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Game.scss';

class Square extends Component {
	constructor(props) {
		super(props);
		this.state = {
			checkedBy : null
		};
	}

	handleClick = () => {
		if (this.props.isFinished || this.state.checkedBy !== null) return;

		this.setState({
			checkedBy : this.props.player
		});

		this.props.onClick(this.props.id);
	};

	render() {
		return (
			<div className="square" onClick={this.handleClick} style={this.props.style}>
				{this.state.checkedBy}
			</div>
		);
	}
}

class Board extends Component {
	renderSquare = (i) => {
		let boxWinStyle;
		if (this.props.isFinished && this.props.winningLine.includes(i)) {
			boxWinStyle = { backgroundColor: 'yellow' };
		}
		return (
			<Square
				id={i}
				onClick={this.props.onSquareClick}
				player={this.props.player}
				isFinished={this.props.isFinished}
				style={boxWinStyle}
			/>
		);
	};

	render() {
		// const player = this.props.player;
		return (
			<React.Fragment>
				<div className="board-row">
					{this.renderSquare(0)}
					{this.renderSquare(1)}
					{this.renderSquare(2)}
				</div>
				<div className="board-row">
					{this.renderSquare(3)}
					{this.renderSquare(4)}
					{this.renderSquare(5)}
				</div>
				<div className="board-row">
					{this.renderSquare(6)}
					{this.renderSquare(7)}
					{this.renderSquare(8)}
				</div>
			</React.Fragment>
		);
	}
}

Board.propTypes = {
	onSquareClick : PropTypes.func,
	player        : PropTypes.string,
	isFinished    : PropTypes.bool,
	winningLine   : PropTypes.array
};

export default Board;
