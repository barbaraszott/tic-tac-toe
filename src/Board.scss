.board {
	margin: 50px 20px;
}

.board-row {
	display: flex;

	&:nth-child(even) {
		border-top: 2px solid #d6e6ea;
		border-bottom: 2px solid #d6e6ea;
	}
}
