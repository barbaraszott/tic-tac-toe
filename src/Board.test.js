import React from 'react';
import { shallow } from 'enzyme';
import Board from './Board';
import renderer from 'react-test-renderer';

const defaultProps = {
	onSquareClick : () => {},
	isFinished    : false,
	winningLine   : [],
	moves         : {},
	winner        : null
};

it('should render without crashing', () => {
	shallow(<Board {...defaultProps} />);
});

it('renders correctly', () => {
	const tree = renderer.create(<Board {...defaultProps} />).toJSON();

	expect(tree).toMatchSnapshot();
});

it('renders correctly winning line [0, 3, 6]', () => {
	const component = shallow(
		<Board
			{...defaultProps}
			isFinished={true}
			winner="x"
			winningLine={[
				0,
				3,
				6
			]}
		/>
	);

	expect(component).toMatchSnapshot();
});

it('should not pass style to squares if there is no winner', () => {
	const component = shallow(<Board {...defaultProps} isFinished={true} />);

	expect(component).toMatchSnapshot();
});
