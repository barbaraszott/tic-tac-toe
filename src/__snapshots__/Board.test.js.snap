// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="board"
>
  <div
    className="board-row"
  >
    <div
      className="square"
      onClick={[Function]}
    >
      <div
        className="square-icon "
      />
    </div>
    <div
      className="square"
      onClick={[Function]}
    >
      <div
        className="square-icon "
      />
    </div>
    <div
      className="square"
      onClick={[Function]}
    >
      <div
        className="square-icon "
      />
    </div>
  </div>
  <div
    className="board-row"
  >
    <div
      className="square"
      onClick={[Function]}
    >
      <div
        className="square-icon "
      />
    </div>
    <div
      className="square"
      onClick={[Function]}
    >
      <div
        className="square-icon "
      />
    </div>
    <div
      className="square"
      onClick={[Function]}
    >
      <div
        className="square-icon "
      />
    </div>
  </div>
  <div
    className="board-row"
  >
    <div
      className="square"
      onClick={[Function]}
    >
      <div
        className="square-icon "
      />
    </div>
    <div
      className="square"
      onClick={[Function]}
    >
      <div
        className="square-icon "
      />
    </div>
    <div
      className="square"
      onClick={[Function]}
    >
      <div
        className="square-icon "
      />
    </div>
  </div>
</div>
`;

exports[`renders correctly winning line [0, 3, 6] 1`] = `
<div
  className="board"
>
  <div
    className="board-row"
  >
    <Square
      id={0}
      isFinished={true}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e7fdff",
        }
      }
    />
    <Square
      id={1}
      isFinished={true}
      onClick={[Function]}
    />
    <Square
      id={2}
      isFinished={true}
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      id={3}
      isFinished={true}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e7fdff",
        }
      }
    />
    <Square
      id={4}
      isFinished={true}
      onClick={[Function]}
    />
    <Square
      id={5}
      isFinished={true}
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      id={6}
      isFinished={true}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e7fdff",
        }
      }
    />
    <Square
      id={7}
      isFinished={true}
      onClick={[Function]}
    />
    <Square
      id={8}
      isFinished={true}
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`should not pass style to squares if there is no winner 1`] = `
<div
  className="board"
>
  <div
    className="board-row"
  >
    <Square
      id={0}
      isFinished={true}
      onClick={[Function]}
    />
    <Square
      id={1}
      isFinished={true}
      onClick={[Function]}
    />
    <Square
      id={2}
      isFinished={true}
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      id={3}
      isFinished={true}
      onClick={[Function]}
    />
    <Square
      id={4}
      isFinished={true}
      onClick={[Function]}
    />
    <Square
      id={5}
      isFinished={true}
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      id={6}
      isFinished={true}
      onClick={[Function]}
    />
    <Square
      id={7}
      isFinished={true}
      onClick={[Function]}
    />
    <Square
      id={8}
      isFinished={true}
      onClick={[Function]}
    />
  </div>
</div>
`;
